File: InnovateConnect - spring/backend/src/main/java/com/innovateconnect/api/models/Application.java
package com.innovateconnect.api.models;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "applications")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class Application {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Integer id;

    @ManyToOne(optional = false)
    @JoinColumn(name = "internship_id", nullable = false)
    private Internship internship;

    @ManyToOne(optional = false)
    @JoinColumn(name = "student_id", nullable = false)
    private Student student;

    @NotBlank(message = "Status is required")
    @Column(name = "status")
    @Builder.Default
    private String status = "Applied"; // Applied, Accepted, Rejected

    @Builder.Default
    @Column(name = "applied_date")
    private LocalDateTime appliedDate = LocalDateTime.now();
}

File: InnovateConnect/InnovateConnect/backend/InnovateConnect.API/Models/Application.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace InnovateConnect.API.Models
{
    public class Application
    {
        public int Id { get; set; }

        public int InternshipId { get; set; }
        [ForeignKey("InternshipId")]
        public Internship? Internship { get; set; }

        public int StudentId { get; set; }
        [ForeignKey("StudentId")]
        public Student? Student { get; set; }

        [Required]
        public string Status { get; set; } = "Applied"; // Applied, Accepted, Rejected

        public DateTime AppliedDate { get; set; } = DateTime.UtcNow;
    }
}
